### PR TITLE
feature/lf-4-import-license

### PR DESCRIPTION
**Pull request by Paurikova2**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  4+5+4+5+8+3+4  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |
## Problem description
Import licenses and license labels in json format into database in correct order.
## Problems
There is problem with new and old license and label ids. We have to ensure correct mapping between old and new data.

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLicenseImportRestController.java [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+     * @throws SQLException
+     * @throws AuthorizeException
+     */
+    @RequestMapping(method = RequestMethod.POST, value = "/labels")
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public ResponseEntity importLicenseLabels(@RequestBody(required = false) List<JsonNode> licenseLabels,
+                                              HttpServletRequest request, HttpServletResponse response)
+            throws SQLException, AuthorizeException {
+
+        if (Objects.isNull(licenseLabels) || licenseLabels.size() < 1) {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLicenseImportRestController.java:79]: Refactor to CollectionUtils.isEmpty

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLicenseImportRestController.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+     * @param request The request object
+     * @return Response entity with status
+     */
+    @RequestMapping(method = RequestMethod.POST, value = "/extendedMapping")
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public ResponseEntity importLicenseLabelExtendedMapping(@RequestBody(required = false) List<JsonNode>
+                                                                        licenseLabelExtendedMappings,
+                                          HttpServletRequest request, HttpServletResponse response) {
+
+        if (Objects.isNull(licenseLabelExtendedMappings) || licenseLabelExtendedMappings.isEmpty()) {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLicenseImportRestController.java:148]: Refactor to CollectionUtils.isEmpty -> it check if the list is null

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLicenseImportRestController.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+     * @throws SQLException
+     * @throws AuthorizeException
+     */
+    @RequestMapping(method = RequestMethod.POST, value = "/licenses")
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public ResponseEntity importLicenses(@RequestBody(required = false) List<JsonNode> licenses,
+                                               HttpServletRequest request, HttpServletResponse response)
+            throws SQLException, AuthorizeException {
+
+        if (Objects.isNull(licenses) || licenses.size() < 1) {
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinLicenseImportRestController.java:226]: CollectionUtils.isEmpty

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/model/ClarinLicenseLabelExtendedMappingRest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.dspace.app.rest.RestResourceController;
+import org.dspace.content.clarin.ClarinLicense;
+
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/model/ClarinLicenseLabelExtendedMappingRest.java:14]: add doc

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/model/LicenseRest.java [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/model/LicenseRest.java:1]: please remove useless changes

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinLabelExtendedMappingRestRepository.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
```
Comment (milanmajchrak) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinLabelExtendedMappingRestRepository.java:9]: Is it used? Could it be removed?

**Review by milanmajchrak — CHANGES_REQUESTED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/model/ClarinLicenseLabelExtendedMappingRest.java (hunk 4 above)
Comment (Paurikova2) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/model/ClarinLicenseLabelExtendedMappingRest.java:14]: This class should be unuseful. I delete it.

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinLabelExtendedMappingRestRepository.java (hunk 6 above)
Comment (Paurikova2) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinLabelExtendedMappingRestRepository.java:9]: It is unuseful. I remove it.

**Pull request merged**